### PR TITLE
Attempt to fix CI failures on re-running checks.

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Use actions/checkout@v2

There is apparently an [issue with GitHub actions/checkout@v1](https://github.com/actions/checkout/issues/23)

### Types of changes:

- [x] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [x] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by kidder** [comment]
@nilsdeppe @nilsleiffischer I tested that this works on #1909

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
@nilsleiffischer could you please either look at this or let us know you're busy so we can get someone else to do it? It would be great to get CI fully working again. Thanks!

File: .github/workflows/Tests.yaml [hunk 1]
```diff
@@ -32,7 +32,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (nilsvu) [inline .github/workflows/Tests.yaml:37]: I think this is the only place in this PR where you need the `fetch-depth: 0`. Please remove the others.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: .github/workflows/Tests.yaml (hunk 1 above)
Comment (nilsdeppe) [reply .github/workflows/Tests.yaml:37]: I'd rather get this merged so we can restart tests... This is also somewhat annoying to debug. I'm also not convinced we don't need it for clang-tidy...

Comment (nilsvu) [reply .github/workflows/Tests.yaml:37]: Sure, I'm fine with merging this now and opening another PR that removes the `fetch-depth`s.

**Pull request merged**